compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -289,13 +287,15 @@ impl Allocation {
     ) -> Result<Allocation<Prov, Extra>, Err> {
         // Compute new pointer provenance, which also adjusts the bytes.
         // Realign the pointer
-        let count128 = (self.bytes.len() / 16) + 1;
-        let mut vec128: Vec<u128> = Vec::with_capacity(count128);
-        vec128.resize(count128, 0);
-        let (buf, _len, _capacity) = vec128.into_raw_parts();
-        let mut vec8: Vec<u8> = unsafe {Vec::from_raw_parts(buf as *mut u8, self.bytes.len(), self.bytes.len())};
-        vec8[..self.bytes.len()].copy_from_slice(&self.bytes);
-        let mut bytes = vec8.into_boxed_slice();
+        let align_usize: usize = self.align.bytes().try_into().unwrap();
+        let count_align = ((self.bytes.len() / align_usize) + 1)*align_usize;
+        
+        let mut vec_align: Vec<u8> = Vec::with_capacity(count_align);
+        vec_align.resize(count_align, 0);
+        assert!(vec_align.as_ptr() as u64 % self.align.bytes() == 0);
```

Comment:
as usize % self.align.bytes_usize()

This is pedantic, but pointers aren't gauranteed to fit in u64, they are gauranteed to fit in usize